iOS/Airbnb/Airbnb/View/DataSource/SearchViewDataSource.swift

```diff
@@ -0,0 +1,116 @@
+        .third([MyHomeSection.ThirdItem].init(repeating: MyHomeSection.ThirdItem.init(image: UIImage(named: "blackHouse") ?? UIImage(), title: "자연생활을 만끽할 수 있는 숙소"), count: 5))
+    ]
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return homeData.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        switch homeData[section] {
+        case let .first(items):
+            return items.count
+        case let .second(items):
+            return items.count
+        case let .third(items):
+            return items.count
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        switch homeData[indexPath.section] {
+        case let .first(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: AdvertisingCell.identifier,
+                                                                for: indexPath) as? AdvertisingCell else {
+                return UICollectionViewCell()
+            }
+            cell.configure(image: items[indexPath.item].image)
+            
+            return cell
+        case let .second(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: NearbyDestinationCell.identifier,
+                                                                for: indexPath) as? NearbyDestinationCell else {
+                return UICollectionViewCell()
+            }
+            let item = items[indexPath.item]
+            cell.configure(image: item.image, title: item.title, distance: item.distance)
+            
+            return cell
+        case let .third(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: LivingSpotCell.identifier,
+                                                                for: indexPath) as? LivingSpotCell else {
+                return UICollectionViewCell()
+            }
+            let item = items[indexPath.item]
+            cell.configure(image: item.image, title: item.title)
+            
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        switch kind {
+        case UICollectionView.elementKindSectionHeader:
+            guard let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: CollectionHeaderView.identifier, for: indexPath) as? CollectionHeaderView else {
+                return UICollectionReusableView()
+            }
+            
+            switch indexPath.section {
+            case 1:
+                headerView.setHeaderText(text: "가까운 여행지 둘러보기")
```

Comment:
뿐만 아니라 의미가 있는 객체를 구성하고 해당 객체에서 필요한 값을 갖는다면
위와 같이 스트링값을 직접 입력하는 방법도 피할 수 있어보입니다.

Reply:
enum 타입으로 조금 더 명확하게 변경해보았습니다. setHeaderText에 직접 입력하는 방법을 좀 더 고민해보겠습니다!
```Swift
let sectionType = Section.allCases[indexPath.section]
            
            switch sectionType {
            case .destination:
                headerView.setHeaderText(text: "가까운 여행지 둘러보기")
                headerView.setHeaderFontSize(size: 22)
            case .livingSpot:
                headerView.setHeaderText(text: "어디에서나,\n여행은 살아보는거야!")
                headerView.setHeaderFontSize(size: 22)
            case .otherwise:
                headerView.setHeaderText(text: "")
            }
```